apps/namegraph.dev/lib/utils.ts

```diff
@@ -1,6 +1,126 @@
 import { clsx, type ClassValue } from "clsx";
 import { twMerge } from "tailwind-merge";
+import {
+  NameGraphGroupedByCategoryResponse,
+  NameGraphGroupingCategory,
+  NameGraphSuggestion,
+} from "@namehash/namegraph-sdk/utils";
+import { createNameGraphClient } from "@namehash/namegraph-sdk";
 
 export function cn(...inputs: ClassValue[]) {
   return twMerge(clsx(inputs));
 }
+
+interface LabelAndModeQueryParams {
+  label: string;
+  mode: "full";
+}
+
+export interface NameGraphRelatedCollection {
+  collection_id: string;
+  collection_title: string;
+  collection_members_count: number;
+}
+
+export interface NameGraphSuggestionCategory {
+  type: NameGraphGroupingCategory;
+  // NameGraph API guarantees "name" will always be a normalized name
+  name: string;
+  // NameGraph API guarantees "suggestions" will never be an empty list
+  suggestions: NameGraphSuggestion[];
+  related_collections: NameGraphRelatedCollection[];
+}
+
+// This array is used for checking when a category is not a "related" category
+const NameGraphSuggestionCategoryTypes = Object.values(
+  NameGraphGroupingCategory,
+);
+
+/**
+ * Common usage NameGraphClient
+ */
+const NameGraphClient = createNameGraphClient();
+
+/**
+ * Main function for fetching name suggestions from the NameGraph API.
+ */
+export const generateNamesByQuery = async (
+  input: LabelAndModeQueryParams,
+): Promise<NameGraphSuggestion[]> => {
+  if (input.label.includes("."))
```

Comment:
Why are we using this strategy? Why aren't we implementing the same strategy as before where if the input includes "." that we internally process that so as to only use the childmost label for our query?

Here we should identify back to the user the exact label that we're using to generate ideas with. Please note the related UX for this that we implemented here: https://www.namerank.io/?name=example.name.with.multiple.labels

![CleanShot 2024-12-20 at 17 29 48](https://github.com/user-attachments/assets/f3006843-a643-42a4-bff9-a57227255e3d)

1. The users raw input query, which may include any number of labels.
2. We show back to the user the label we're actually going to use in the query.
3. We show the result of the query.

Please study the UX and implementation referenced above. We should follow the same pattern here.

Reply:
Now following the same pattern of using the childmost label for search. Mini-app is also displaying both the complete and the query-used strings, alongside with the results!